tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -23,22 +23,22 @@ using namespace shogun;
 #include <shogun/mathematics/eigen3.h>
 using namespace Eigen;
 
-//function to create sparse matrix
-template<typename T>
-SGSparseMatrix<T> GenerateSparseMatrix(const int32_t numberOfFeatures,
-                                       const int32_t numberOfVectors,
-                                       const float64_t sparseLevel)
+//function can generate both sparse and dense matrix accroding to sparse convention
+template<class MatrixType>
+void GenerateMatrixAccrodingToSparseConvention(const float64_t &sparseLevel,
+                                               const int32_t   &numberOfFeatures,
+                                               const int32_t   &numberOfVectors,
+                                               const int32_t   &randSeed,
```

Comment:
Why are all these const and passed by reference? I mean, the pass by reference is correct, but there is no gain here since the arguments are just primitive and very simple data types.